src/ui/main.cpp

```diff
@@ -52,123 +48,181 @@
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 2);
+    glslVersionOut = "#version 150";
 #else
-    // GL 3.0 + GLSL 130
-    const char* glsl_version = "#version 130";
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_FLAGS, 0);
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
-    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
-    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 0);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, major);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, minor);
+    glslVersionOut = "#version 330 core";
+#endif
 #endif
 
-    // Create window with graphics context
     SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
     SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
     SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
-    SDL_DisplayMode current;
-    SDL_GetCurrentDisplayMode(0, &current);
-    const auto window_flags = static_cast<SDL_WindowFlags>(SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI);
-    SDLState   sdlState;
-    sdlState.window = SDL_CreateWindow("opendigitizer UI", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1280, 720, window_flags);
-    if (!sdlState.window) {
-        std::print(stderr, "Failed to create SDL window: {}!\n", SDL_GetError());
-        return 1;
+
+#ifdef __APPLE__
+    constexpr SDL_WindowFlags windowFlags = SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_HIGH_PIXEL_DENSITY | SDL_WINDOW_ALLOW_HIGHDPI;
+#else
+    constexpr SDL_WindowFlags windowFlags = SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_HIGH_PIXEL_DENSITY;
+#endif
+    g_Window = SDL_CreateWindow("OpenDigitizer UI", 1280, 720, windowFlags);
+    if (!g_Window) {
+        std::println(stderr, "[Main] SDL_CreateWindow failed: '{}'", SDL_GetError());
+        return false;
     }
-    sdlState.glContext = SDL_GL_CreateContext(sdlState.window);
-    if (!sdlState.glContext) {
-        std::print(stderr, "Failed to initialize GL context!\n");
-        return 1;
+
+    g_GLContext = SDL_GL_CreateContext(g_Window);
+    if (!g_GLContext) {
+        std::println(stderr, "[Main] SDL_GL_CreateContext({}.{}) failed: '{}'", major, minor, SDL_GetError());
+        SDL_DestroyWindow(g_Window);
+        g_Window = nullptr;
+        return false;
+    }
+    std::println("[Main] GL_VERSION:   {}", reinterpret_cast<const char*>(glGetString(GL_VERSION)));
+    std::println("[Main] GL_RENDERER:  {}", reinterpret_cast<const char*>(glGetString(GL_RENDERER)));
+    std::println("[Main] GLSL_VERSION: {}", reinterpret_cast<const char*>(glGetString(GL_SHADING_LANGUAGE_VERSION)));
+
+    std::println("[Main] OpenGL context successfully created using shader '{}'", glslVersionOut);
+    return true;
+}
+
+bool initSDL(std::string& glslVersion) {
+    if (!SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO)) {
```

Comment:
```suggestion
    if (!SDL_Init(SDL_INIT_VIDEO)) {
```

If audio is not used we should not initialize it here. Otherwise we have to make sure that at least the dependencies for one audio backend (libpulseaudio, alsa, etc) are available or this call will fail if we use our own build of SDL.